Gems/SurfaceData/Code/CMakeLists.txt

```diff
@@ -70,6 +70,7 @@ if (PAL_TRAIT_BUILD_HOST_TOOLS)
                 AZ::AzToolsFramework
                 Gem::SurfaceData.Static
                 Gem::LmbrCentral.Editor
+                Gem::Terrain
         RUNTIME_DEPENDENCIES
             Gem::LmbrCentral.Editor
```

Comment:
Doesn't the terrain gem already depend on the SurfaceData gem as a build dependency?
I guess this works by accident, in that they are both MODULEs and thus wont try to link to each other, but its usually poor form for a build dependency to be circular. 

An alternative would be to export the public API of Terrain as a INTERFACE or STATIC (header only or staticlib) and have both modules (Terrain and SurfaceData) depend on it.

Its not important to fix right now if it works (esp in linux), but we probably need to figure out a 'standard' to move towards when gems are interdependent on each other API wise.

Also, shoudln't the editor / tools module depend on the terrain editor/tools module?  Not the runtime?  This will likely cause "Terrain.dll" and "Terrain.Editor.dll", and "SurfaceData.Editor.dll" modules to be loaded if both the Terrain and SurfaceData  gems are enabled.

Reply:
This dependency should definitely not be added.  SurfaceData should NOT require Terrain, that's the whole point behind this PR.

Reply:
Removed